tfhe/src/integer/ciphertext/mod.rs

```diff
@@ -1,490 +1,11 @@
-    fn holds_boolean_value(&self) -> bool {
-        self.blocks()[0].degree.get() <= 1
-            && self.blocks()[1..]
-                .iter()
-                .all(|block| block.degree.get() == 0)
-    }
-
-    fn into_blocks(self) -> Vec<Ciphertext>;
-}
-
-impl IntegerCiphertext for RadixCiphertext {
-    fn blocks(&self) -> &[Ciphertext] {
-        &self.blocks
-    }
-
-    fn from_blocks(blocks: Vec<Ciphertext>) -> Self {
-        Self::from(blocks)
-    }
-
-    fn blocks_mut(&mut self) -> &mut [Ciphertext] {
-        &mut self.blocks
-    }
-}
-
-impl IntegerRadixCiphertext for RadixCiphertext {
-    const IS_SIGNED: bool = false;
-
-    fn into_blocks(self) -> Vec<Ciphertext> {
-        self.blocks
-    }
-}
-
-impl IntegerCiphertext for SignedRadixCiphertext {
-    fn blocks(&self) -> &[Ciphertext] {
-        &self.blocks
-    }
-
-    fn from_blocks(blocks: Vec<Ciphertext>) -> Self {
-        Self::from(blocks)
-    }
-
-    fn blocks_mut(&mut self) -> &mut [Ciphertext] {
-        &mut self.blocks
-    }
-}
-
-impl IntegerRadixCiphertext for SignedRadixCiphertext {
-    const IS_SIGNED: bool = true;
-
-    fn into_blocks(self) -> Vec<Ciphertext> {
-        self.blocks
-    }
-}
-
-impl IntegerCiphertext for CrtCiphertext {
-    fn blocks(&self) -> &[Ciphertext] {
-        &self.blocks
-    }
-
-    fn from_blocks(blocks: Vec<Ciphertext>) -> Self {
-        let moduli = blocks.iter().map(|x| x.message_modulus.0 as u64).collect();
-        Self { blocks, moduli }
-    }
-
-    fn blocks_mut(&mut self) -> &mut [Ciphertext] {
-        &mut self.blocks
-    }
-
-    fn moduli(&self) -> Vec<u64> {
-        self.moduli.clone()
-    }
-}
-
-/// Structure containing a ciphertext in CRT decomposition.
-///
-/// For this CRT decomposition, each block is encrypted using
-/// the same parameters.
-#[derive(Serialize, Clone, Deserialize)]
-pub struct BaseCrtCiphertext<Block> {
-    pub(crate) blocks: Vec<Block>,
-    pub(crate) moduli: Vec<u64>,
-}
-
-/// Structure containing a ciphertext in CRT decomposition.
-pub type CrtCiphertext = BaseCrtCiphertext<Ciphertext>;
-
-/// Structure containing a **compressed** ciphertext in CRT decomposition.
-pub type CompressedCrtCiphertext = BaseCrtCiphertext<CompressedCiphertext>;
-
-impl<Block> From<(Vec<Block>, Vec<u64>)> for BaseCrtCiphertext<Block> {
-    fn from((blocks, moduli): (Vec<Block>, Vec<u64>)) -> Self {
-        Self { blocks, moduli }
-    }
-}
-
-impl From<CompressedCrtCiphertext> for CrtCiphertext {
-    fn from(compressed: CompressedCrtCiphertext) -> Self {
-        let blocks = compressed
-            .blocks
-            .into_iter()
-            .map(From::from)
-            .collect::<Vec<_>>();
-        let moduli = compressed.moduli;
-        Self::from((blocks, moduli))
-    }
-}
+mod compact_list;
+mod compressed;
+mod utils;
+
+pub use base::*;
+pub use boolean_value::*;
+pub use compact_list::*;
+pub use compressed::*;
+pub use utils::*;
```

Comment:
pub use the minimum required/which makes sense to compile

Reply:
Removed